plugins/generate-proptypes/ts-utils.mjs

```diff
@@ -0,0 +1,30 @@
+import * as pathModule from "path";
+import fg from "fast-glob";
+import * as ttp from "typescript-to-proptypes";
+
+/**
+ * Generate a TypeScript program object for all the ts files in src
+ */
+async function generateTsProgram() {
+  const tsconfig = ttp.loadConfig(
+    pathModule.resolve(pathModule.resolve(), "./tsconfig.json")
+  );
+
+  const tsFiles = await fg("**/*.{ts,tsx}", {
+    absolute: true,
+    cwd: pathModule.resolve(pathModule.resolve(), "./src"),
+  });
+
+  return ttp.createProgram(tsFiles, tsconfig);
+}
+
+/**
+ * Generate a PropTypes AST from the TypeScript interfaces in the current file
+ * @param filename File to get the prop types from
+ * @param program TypeScript program object
+ */
+function generateProptypes(filename, program) {
+  return ttp.parseFromProgram(filename, program);
+}
+
+export { generateTsProgram, generateProptypes };
```

Comment:
That would mean we would have to await the result of `generateTsProgram()` potentially multiple times per file, rather than just once, unless we cached the result or something.

I also think the code is cleaner the way I've done it, as each function is concerned with a single main task. So I'll leave it as it is for this one 👍🏻

Reply:
Yes I agree, that's why I gave the quick (but admittedly quite ugly) code sample of how we could cache it.

Which is cleaner is a matter of opinion, happy to stick with your way as I don't feel that strongly about it.